File: app/src/main/kotlin/io/mochadwi/util/service/NotifyDailyWorker.java
package io.mochadwi.util.service;

import android.content.Context;
import android.content.Intent;
import android.util.Log;

import org.joda.time.DateTime;

import androidx.annotation.NonNull;
import androidx.work.Worker;
import androidx.work.WorkerParameters;

import static io.mochadwi.util.helper.AppHelper.Const.TAG_MOVIE_DAILY;

public class NotifyDailyWorker extends Worker {

    private static final String TAG = NotifyDailyWorker.class.getSimpleName();

    public NotifyDailyWorker(@NonNull Context context, @NonNull WorkerParameters workerParams) {
        super(context, workerParams);
    }

    @NonNull
    @Override
    public Result doWork() {
        if (new DateTime().getHourOfDay() == 7) {
            getApplicationContext().startService(new Intent(getApplicationContext(), NotificationDailyService.class));
            Log.d(TAG_MOVIE_DAILY, "worker daily running");

            Log.d(TAG_MOVIE_DAILY, "worker success");
            return Result.success();
        } else {
            Log.d(TAG_MOVIE_DAILY, "worker retry");
            return Result.retry();
        }
    }
}

File: app/src/main/kotlin/io/mochadwi/util/service/NotifyDailyWorker.kt
package io.mochadwi.util.service

import android.content.Context
import android.content.Intent
import android.util.Log
import androidx.work.Worker
import androidx.work.WorkerParameters
import io.mochadwi.util.helper.AppHelper.Const.TAG_MOVIE_DAILY
import org.joda.time.DateTime

class NotifyDailyWorker(context: Context, workerParams: WorkerParameters) : Worker(context, workerParams) {

    override fun doWork(): Result {
        return when (DateTime().hourOfDay) {
            7 -> {
                applicationContext.startService(Intent(applicationContext, NotificationDailyService::class.java))
                Log.d(TAG_MOVIE_DAILY, "worker daily running")
                Result.success()
            }
            else -> {
                Log.d(TAG_MOVIE_DAILY, "worker retry")
                Result.retry()
            }
        }
    }

    companion object {

        private val TAG = NotifyDailyWorker::class.java.simpleName
    }
}
